### PR TITLE
AHOYAPPS-743: Use AppScreenSource from SDK for in-app screen capture

### PR DESCRIPTION
**Pull request by timrozum**
Use `AppScreenSource` from SDK instead of using `RPScreenRecorder` directly. 

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [X] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -1,9 +1,10 @@
+source 'git@github.com:twilio/cocoapod-specs-internal.git'
 source 'https://github.com/CocoaPods/Specs'
 
 workspace 'VideoQuickStart'
 
 abstract_target 'TwilioVideo' do
-  pod 'TwilioVideo', '~> 3.7'
+  pod 'TwilioVideo', '3.7.1-9e1f62ff'
```
Comment (timrozum) [inline Podfile:7]: These changes will need to be updated before merge.

File: ReplayKitExample/ReplayKitExample/ViewController.swift [hunk 2]
```diff
@@ -254,39 +254,36 @@ class ViewController: UIViewController {
     }
 
     private func stopConference(error: Error?) {
-        // Stop recording the screen.
-        let recorder = RPScreenRecorder.shared()
-        recorder.stopCapture { (captureError) in
-            if let error = captureError {
-                print("Screen capture stop error: ", error as Any)
+        appScreenSource?.stopCapture { captureError in
+            if let captureError = captureError {
```
Comment (paynerc) [inline ReplayKitExample/ReplayKitExample/ViewController.swift:258]: The completion is fired on the main thread, correct?

File: ReplayKitExample/ReplayKitExample/ViewController.swift [hunk 3]
```diff
@@ -254,39 +254,36 @@ class ViewController: UIViewController {
...
-                    if let userError = error {
-                        self.infoLabel?.text = userError.localizedDescription
-                    }
-                }
+            if let userError = error {
+                self.infoLabel?.text = userError.localizedDescription
             }
         }
-
+
```
Comment (paynerc) [inline ReplayKitExample/ReplayKitExample/ViewController.swift:286]:
```suggestion

```

File: ReplayKitExample/ReplayKitExample/ViewController.swift [hunk 4]
```diff
@@ -307,71 +304,48 @@ class ViewController: UIViewController {
...
-                    self.spinner.startAnimating()
-                    self.infoLabel?.isHidden = true
-                    self.connectToRoom(name: "conference", encodingParameters: encodingParams)
-
-                    let playVideo = UIBarButtonItem(title: "Play Video", style: .plain, target: self, action: #selector(self.pickDocument(_:)))
-                    let browseWeb = UIBarButtonItem(title: "Browse Web", style: .plain, target: self, action: #selector(self.browseWeb(_:)))
-                    self.navigationItem.leftBarButtonItem = playVideo
-                    self.navigationItem.rightBarButtonItem = browseWeb
-                }
+            self.conferenceButton?.isEnabled = true
```
Comment (paynerc) [inline ReplayKitExample/ReplayKitExample/ViewController.swift:323]: And completion here is also on the main thread, correct?

File: ReplayKitExample/ReplayKitExample/ViewController.swift [hunk 5]
```diff
@@ -421,6 +395,13 @@ class ViewController: UIViewController {
         // Connect to the Room using the options we provided.
         conferenceRoom = TwilioVideoSDK.connect(options: connectOptions, delegate: self)
     }
+    
+    private func handleAppScreenSourceAvailabilityChange() {
+        // Assume we will get an error raised if we are actively broadcasting / capturing and access is "stolen".        }
```
Comment (paynerc) [inline ReplayKitExample/ReplayKitExample/ViewController.swift:400]:
```suggestion
        // Assume we will get an error raised if we are actively broadcasting / capturing and access is "stolen".
```

**Review by paynerc — CHANGES_REQUESTED** [review]
Looks good! Minor fixes and two questions.

File: ReplayKitExample/ReplayKitExample/ViewController.swift (hunk 2 above)
Comment (timrozum) [reply ReplayKitExample/ReplayKitExample/ViewController.swift:258]: Correct always.

File: ReplayKitExample/ReplayKitExample/ViewController.swift (hunk 4 above)
Comment (timrozum) [reply ReplayKitExample/ReplayKitExample/ViewController.swift:323]: Yes so much easier now. 😄

**Review by paynerc — APPROVED** [review]

**Comment by timrozum** [comment]
We need to have a public SDK release with in-app screen capture included before this can be merged.

**Review by piyushtank — APPROVED** [review]

**Pull request merged**